src/Consolonia.Core/Drawing/PixelBufferImplementation/PixelBuffer.cs

```diff
@@ -86,15 +86,21 @@ public string PrintBuffer()
 
             for (ushort j = 0; j < Height; j++)
             {
-                for (ushort i = 0; i < Width; i++)
+                for (ushort i = 0; i < Width;)
                 {
-                    if (i == Width - 1 && j == Height - 1)
-                        break;
                     Pixel pixel = this[new PixelBufferCoordinate(i, j)];
-                    string text = pixel.IsCaret() ? "Ꮖ" : pixel.Foreground.Symbol.GetText();
-
-                    //todo: check why cursor is not drawing
-                    stringBuilder.Append(text);
+                    if (pixel.Width > 0)
+                    {
+                        string text = pixel.IsCaret() ? "Ꮖ" : pixel.Foreground.Symbol.GetText();
+
+                        //todo: check why cursor is not drawing
+                        stringBuilder.Append(text);
+                        i += pixel.Width;
```

Comment:
What would be width for Caret?

Reply:
The width of the caret comes from the symbol used, so this should be good.